### PR TITLE
Selenium 3 Final?

### PR DESCRIPTION
**Pull request by titusfortner**
I think I incorporated everything from last time, but...
This is rebased to latest, and it includes a bunch of refactoring for method/line/class/module sizes, etc, which might be more than we want to do right now, but I figured if we were ever going to change things now would be a good time with a major version change.

Also, I actually like the idea of maintaining the rubocop file to ensure future consistency.

This passes all the specs on mac, but I haven't had a chance to test it on all the platforms yet, but I didn't want to wait until after I got back from vacation before we start looking at it.

### CODE REVIEW

File: rb/lib/selenium/webdriver/common/w3c_error.rb [hunk 1]
```diff
@@ -71,7 +70,7 @@ class InvalidSelectorError < WebDriverError; end
 
       #
       # Occurs if the given session id is not in the list of active sessions,
-      # meaning the session either does not exist or that it’s not active.
+      # meaning the session either does not exist or that it''s not active.
```
Comment (p0deje) [inline rb/lib/selenium/webdriver/common/w3c_error.rb]: single quote added twice

File: rb/lib/selenium/webdriver/common/w3c_error.rb [hunk 2]
```diff
@@ -83,7 +82,7 @@ class InvalidSessionIdError < WebDriverError; end
       class JavascriptError < WebDriverError; end
 
       #
-      # The target for mouse interaction is not in the browser’s viewport and
+      # The target for mouse interaction is not in the browser''s viewport and
```
Comment (p0deje) [inline rb/lib/selenium/webdriver/common/w3c_error.rb]: single quote added twice

File: rb/lib/selenium/webdriver/common/w3c_error.rb [hunk 3]
```diff
@@ -142,7 +141,7 @@ class StaleElementReferenceError < WebDriverError; end
       class TimeoutError < WebDriverError; end
 
       #
-      # A request to set a cookie’s value could not be satisfied.
+      # A request to set a cookie''s value could not be satisfied.
```
Comment (p0deje) [inline rb/lib/selenium/webdriver/common/w3c_error.rb]: single quote added twice

**Comment by p0deje** [comment]
@titusfortner Looks really amazing, I'm thrilled to merge this as soon as possible. Great work!

**Comment by titusfortner** [comment]
Really, "single quote added twice" is the only thing you found that should get fixed? :)
I haven't had a chance to test it on Windows or Linux yet. I'm going to see if I can get a VM running on my laptop to do so.
Either way, best to get it merged, then test through the beta. :)

**Comment by p0deje** [comment]
Don't forget that I only reviewed the changes, not tested them :)

I can test it if you can't get a VM running :)

**Pull request closed without merging**